docs/contributing.rst

```diff
@@ -4,150 +4,110 @@
+     either in a GitHub issue, on the discussion forum, or on IRC. Making sure
+     your ideas and solutions are aligned with other contributors greatly
+     increase the odds of your pull request being quickly accepted.
 
-   To run tests with test coverage statistics::
+#. Create a new branch, based on the ``develop`` branch, for every feature or
+   bug fix. Keep branches small and on topic, as that makes them far easier to
+   review. We often use the following naming convention for branches:
 
-       py.test --cov=mopidy --cov-report=term-missing
-
-   Test coverage statistics can also be viewed online at
-   `coveralls.io <https://coveralls.io/r/mopidy/mopidy>`_.
-
-#. Always check the code for errors and style issues using flake8::
-
-       flake8
-
-   If successful, the command will not print anything at all. Ignore the rare
-   cases you need to ignore a check use `# noqa: <code>` so we can lookup what
-   you are ignoring.
-
-#. Finally, there is the ultimate but a bit slower command. To run both tests,
-   docs build, and flake8 linting, run::
-
-       tox
-
-   This will run exactly the same tests as `Travis CI
-   <https://travis-ci.org/mopidy/mopidy>`_ runs for all our branches and pull
-   requests. If this command turns green, you can be quite confident that your
-   pull request will get the green flag from Travis as well, which is a
-   requirement for it to be merged.
-
-
-Submitting changes
-==================
-
-- One branch per feature or fix. Keep branches small and on topic.
-
-- Follow the :ref:`code style <codestyle>`, especially make sure ``flake8``
-  does not complain about anything.
+   - Features get the prefix ``feature/``
 
-- Write good commit messages. Here's three blog posts on how to do it right:
+   - Bug fixes get the prefix ``fix/``
 
-  - `Writing Git commit messages
-    <http://365git.tumblr.com/post/3308646748/writing-git-commit-messages>`_
+   - Improvements to the documentation get the prefix ``docs/``
 
-  - `A Note About Git Commit Messages
-    <http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_
+#. Follow the :ref:`code style <codestyle>`, especially make sure the
+   ``flake8`` linter does not complain about anything. Travis CI will check
+   that your pull request is "flake8 clean". See :ref:`code-linting`.
 
-  - `On commit messages
-    <http://who-t.blogspot.ch/2009/12/on-commit-messages.html>`_
+#. Include tests for any new feature or substantial bug fix. See
+   :ref:`running-tests`.
 
-- Send a pull request to the ``develop`` branch. See the `GitHub pull request
-  docs <https://help.github.com/articles/using-pull-requests>`_ for help.
+#. Include documentation for any new feature. See :ref:`writing-docs`.
 
+#. Feel free to include a changelog entry in your pull request. The changelog
+   is in :file:`docs/changelog.rst`.
 
-Additional resources
-====================
+#. Write good commit messages. Here's three blog posts on how to do it right:
```

Comment:
We need to add something about our `topic: message...` convention at least, we can't expect people to read all the external links. I.e. we need a tl;dr.